### PR TITLE
chore(graphql): Use element template generator for GraphQL outbound c…

### PR DESCRIPTION
**Pull request by sbuettner**
…onnector

## Related issues

closes camunda/team-connectors/issues/616

### CODE REVIEW

**Comment by sbuettner** [comment]
@Oleksiivanov Yeah, its a preview. I want to allow to review the old one against the new one. Will remove it before switching to review.

**Review by markfarkas-camunda — COMMENTED** [review]
Looks good so far! 👍

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json [hunk 1]
```diff
@@ -0,0 +1,469 @@
+{
+  "$schema" : "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
+  "name" : "GraphQL Outbound Connector",
+  "id" : "io.camunda.connectors.GraphQL.v1",
+  "description" : "Execute GraphQL query",
+  "documentationRef" : "https://docs.camunda.io/docs/components/connectors/protocol/graphql/",
+  "version" : 5,
```
Comment (igpetrov) [inline connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:7]: Does hybrid need to be version 5?

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+    }, {
+      "name" : "None",
+      "value" : "noAuth"
+    }, {
+      "name" : "OAuth 2.0",
+      "value" : "oauth-client-credentials-flow"
+    } ]
+  }, {
+    "id" : "authentication.apiKeyLocation",
+    "label" : "Api key location",
```
Comment (igpetrov) [inline connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:79]:
```suggestion
    "label" : "API key location",
```

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+    },
+    "condition" : {
+      "property" : "authentication.type",
+      "equals" : "oauth-client-credentials-flow",
+      "type" : "simple"
+    },
+    "type" : "String"
+  }, {
+    "id" : "authentication.clientId",
+    "label" : "Client id",
```
Comment (igpetrov) [inline connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:221]:
```suggestion
    "label" : "Client ID",
```

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+    "feel" : "required",
+    "group" : "endpoint",
+    "binding" : {
+      "name" : "graphql.headers",
+      "type" : "zeebe:input"
+    },
+    "type" : "String"
+  }, {
+    "id" : "graphql.query",
+    "label" : "Query/Mutation",
```
Comment (igpetrov) [inline connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:372]: No sure if my suggestion is right tho 🤔 

```suggestion
    "label" : "Query/mutation",
```

File: connectors/http/graphql/element-templates/graphql-outbound-connector.json [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+    }, {
+      "name" : "None",
+      "value" : "noAuth"
+    }, {
+      "name" : "OAuth 2.0",
+      "value" : "oauth-client-credentials-flow"
+    } ]
+  }, {
+    "id" : "authentication.apiKeyLocation",
+    "label" : "Api key location",
```
Comment (igpetrov) [inline connectors/http/graphql/element-templates/graphql-outbound-connector.json:74]: Here and others similar cases mentioned in the hybrid one.

Also, probably worth checking code-based generation for correct labels too.

```suggestion
    "label" : "API key location",
```

**Review by igpetrov — APPROVED** [review]
LGTM, some styling comments.

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json (hunk 1 above)
Comment (sbuettner) [reply connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:7]: Its coming from the generator. I dont think we support custom version per generated output cc @chillleader

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json (hunk 2 above)
Comment (sbuettner) [reply connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:79]: Its based on the base http library to aligned with the rest connector. I would not change the template as its generated. We can fix it in the common library in another PR.

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json (hunk 3 above)
Comment (sbuettner) [reply connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:221]: Same as above.

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json (hunk 4 above)
Comment (sbuettner) [reply connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:372]: Again, we need to fix it in the code, not the element template.

File: connectors/http/graphql/element-templates/graphql-outbound-connector.json (hunk 5 above)
Comment (sbuettner) [reply connectors/http/graphql/element-templates/graphql-outbound-connector.json:74]: see above

File: connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json (hunk 2 above)
Comment (igpetrov) [reply connectors/http/graphql/element-templates/graphql-outbound-connector-hybrid.json:79]: Thank you for clarification!

File: connectors/http/graphql/src/main/java/io/camunda/connector/http/graphql/model/GraphQLRequest.java [hunk 6]
```diff
@@ -6,50 +6,71 @@
...
-    return "GraphQLRequest{" + "query='" + query + '\'' + ", variables=" + variables + '}';
-  }
+  public record GraphQL(
+      @TemplateProperty(
+              id = "query",
+              label = "Query/Mutation",
+              description =
+                  "See <a href=\"https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/graphql/#querymutation\" target=\"_blank\">documentation</a>",
+              type = TemplateProperty.PropertyType.Text,
+              // TODO add support for language property supported by element templates: language:
```
[the remark this answers is not in the document]
Comment (markfarkas-camunda) [reply connectors/http/graphql/src/main/java/io/camunda/connector/http/graphql/model/GraphQLRequest.java:38]: @sbuettner it is used for auto resizing

[the remark this answers is not in the document]
Comment (sbuettner) [reply connectors/http/graphql/src/main/java/io/camunda/connector/http/graphql/model/GraphQLRequest.java:38]: @markfarkas-camunda I added it to the list of improvements for the element template generator: https://github.com/orgs/camunda/projects/23/views/1?filterQuery=chill&pane=issue&itemId=51002405

**Pull request merged**